api/src/pcapi/core/offers/repository.py

```diff
@@ -308,15 +309,19 @@ def get_collective_offers_by_filters(
         match status:
             # Status BOOKED == offer is sold out and last booking link to this reservation is not PENDING
             case educational_models.CollectiveOfferDisplayedStatus.BOOKED.value:
-                booking_subquery = (
-                    educational_models.CollectiveStock.query.with_entities(
-                        educational_models.CollectiveStock.collectiveOfferId
-                    )
-                    .join(educational_models.CollectiveBooking, educational_models.CollectiveStock.collectiveBookings)
-                    .filter(
-                        educational_models.CollectiveBooking.status
-                        != educational_models.CollectiveBookingStatus.PENDING
-                    )
+                booking_subquery = educational_models.CollectiveStock.query.with_entities(
+                    educational_models.CollectiveStock.collectiveOfferId
+                ).join(
+                    educational_models.CollectiveBooking,
+                    and_(
+                        educational_models.CollectiveStock.id == educational_models.CollectiveBooking.collectiveStockId,
+                        educational_models.CollectiveBooking.status.notin_(
```

Comment:
Détail : il me semble que @dbaty avait un jour homogénéisé le code avec `not_in` : 
```suggestion
                        educational_models.CollectiveBooking.status.not_in(
```